### PR TITLE
[AlwaysOn] Convert sample to Kotlin

### PR DESCRIPTION
**Pull request by alexvanyo**
Updates `AlwaysOn` from Java to Kotlin. I split this up into a bunch of smaller commits that slow change the existing sample to a refreshed one.

Highlighted changes:
- The first commit performs an automatic conversion from Java to Kotlin
- `minApi` updated to `26`, and switched to using `java.time.*` APIs for the time and interval handling
- Switch from using `Handler` for `active` mode updates to a coroutine-based method
- `core-ktx` brought in for a couple of useful extensions
- `findViewById` replaced by `ViewBinding`

### CODE REVIEW

File: AlwaysOn/Wearable/src/main/java/com/example/android/wearable/wear/alwayson/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+             *
+             * Since we're potentially applying negative padding, we have ensured
+             * that the containing view is sufficiently padded (see res/layout/activity_main.xml).
+             *
+             * Activities should also avoid solid white areas to prevent pixel burn-in. Both of
+             * these requirements only apply in ambient mode, and only when this property is set
+             * to true.
+             */
+            if (doBurnInProtection) {
+                binding.container.updatePadding(
```
Comment (jdkoren) [inline AlwaysOn/Wearable/src/main/java/com/example/android/wearable/wear/alwayson/MainActivity.kt:311]: In my experience, it's weird to set negative padding values. What do you think about setting `translationX` and/or `translationY` instead?

Comment (alexvanyo) [reply AlwaysOn/Wearable/src/main/java/com/example/android/wearable/wear/alwayson/MainActivity.kt]: Agreed, updated in latest commit.

File: AlwaysOn/Wearable/src/main/java/com/example/android/wearable/wear/alwayson/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+
+/**
+ * IMPORTANT NOTE: Most apps shouldn't use always on ambient mode, as it drains battery life. Unless
+ * required, it's much better to allow the system to take over after the user stops interacting
+ * with your app.
+ *
+ * Demonstrates support for *Ambient Mode* by attaching ambient mode support to the activity,
+ * and listening for ambient mode updates (onEnterAmbient, onUpdateAmbient, and onExitAmbient) via a
+ * named AmbientCallback subclass.
+ *
```
[the remark this answers is not in the document]
Comment (alexvanyo) [reply AlwaysOn/Wearable/src/main/java/com/example/android/wearable/wear/alwayson/MainActivity.kt:50]: Remove the extra line breaks and standardized the KDoc formatting in the latest commit.

File: AlwaysOn/Wearable/src/main/java/com/example/android/wearable/wear/alwayson/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright (C) 2015 Google Inc. All Rights Reserved.
```
Comment (codingjeremy) [inline AlwaysOn/Wearable/src/main/java/com/example/android/wearable/wear/alwayson/MainActivity.kt:2]: Minor nit, you should update the copyrights to 2021.

**Review by codingjeremy — APPROVED** [review]
Great job!

I only had a nit on the copyrights being 2015, I would set to 2021.

File: AlwaysOn/Wearable/src/main/java/com/example/android/wearable/wear/alwayson/MainActivity.kt (hunk 3 above)
Comment (alexvanyo) [reply AlwaysOn/Wearable/src/main/java/com/example/android/wearable/wear/alwayson/MainActivity.kt]: Updated/added source headers for everything in `AlwaysOn`

**Comment by codingjeremy** [comment]
LGTM

**Review by jdkoren — APPROVED** [review]

**Pull request merged**